src/Umbraco.Infrastructure/Migrations/MigrationContext.cs

```diff
@@ -8,13 +10,15 @@ namespace Umbraco.Cms.Infrastructure.Migrations;
 /// </summary>
 internal class MigrationContext : IMigrationContext
 {
+    private readonly Action _onDoneAction;
     private readonly List<Type> _postMigrations = new();
 
     /// <summary>
     ///     Initializes a new instance of the <see cref="MigrationContext" /> class.
     /// </summary>
-    public MigrationContext(MigrationPlan plan, IUmbracoDatabase? database, ILogger<MigrationContext> logger)
+    public MigrationContext(MigrationPlan plan, IUmbracoDatabase? database, ILogger<MigrationContext> logger, Action onDoneAction)
     {
+        _onDoneAction = onDoneAction;
```

Comment:
This should also be renamed (completeAction?). Same with `OnMigrationDone` method in class below and the comment referring to `OnDone`.

Maybe this last parameter can be optional? Then you don't have to pass in nop action everywhere in tests.